src/App.js

```diff
@@ -1,86 +1,93 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { TodoList } from './components/TodoList';
+import { NewTodo } from './components/NewTodo';
+import { TodosFilter } from './components/TodosFilter';
 
 function App() {
-  return (
-    <section className="todoapp">
-      <header className="header">
-        <h1>todos</h1>
+  const [todos, setTodos] = useState([]);
+  const [toggleStatus, setToggle] = useState(true);
+  const [selectedFilter, setFilter] = useState('All');
 
-        <form>
-          <input
-            type="text"
-            className="new-todo"
-            placeholder="What needs to be done?"
-          />
-        </form>
-      </header>
+  const getCountNotCompleted = () => (
+    todos.filter(todo => !todo.completed).length
+  );
 
-      <section className="main">
-        <input type="checkbox" id="toggle-all" className="toggle-all" />
-        <label htmlFor="toggle-all">Mark all as complete</label>
+  const countNotCompleted = getCountNotCompleted();
 
-        <ul className="todo-list">
-          <li>
-            <div className="view">
-              <input type="checkbox" className="toggle" />
-              <label>asdfghj</label>
-              <button type="button" className="destroy" />
-            </div>
-            <input type="text" className="edit" />
-          </li>
+  const toggleStatusAll = () => {
+    setTodos(todos.map(todo => (
+      { ...todo, completed: toggleStatus }
+    )));
+    setToggle(!toggleStatus);
+  };
 
-          <li className="completed">
-            <div className="view">
-              <input type="checkbox" className="toggle" />
-              <label>qwertyuio</label>
-              <button type="button" className="destroy" />
-            </div>
-            <input type="text" className="edit" />
-          </li>
+  const toggleStatusItem = (itemId) => {
+    setTodos(todos.map((todo) => {
+      if (todo.id === itemId) {
+        return { ...todo, completed: !todo.completed };
+      }
 
-          <li className="editing">
-            <div className="view">
-              <input type="checkbox" className="toggle" />
-              <label>zxcvbnm</label>
-              <button type="button" className="destroy" />
-            </div>
-            <input type="text" className="edit" />
-          </li>
+      return todo;
+    }));
+  };
 
-          <li>
-            <div className="view">
-              <input type="checkbox" className="toggle" />
-              <label>1234567890</label>
-              <button type="button" className="destroy" />
-            </div>
-            <input type="text" className="edit" />
-          </li>
-        </ul>
-      </section>
+  const filterTodos = () => {
+    const FILTERS = {
+      All: todos,
+      Active: todos.filter(todo => !todo.completed),
+      Completed: todos.filter(todo => todo.completed),
+    };
```

Comment:
```suggestion
    const filterHash = {
      [FILTERS.all]: () => todos,
      [FILTERS.active]: () => todos.filter(todo => !todo.completed),
      [FILTERS.completed]: () => todos.filter(todo => todo.completed),
    };
```
до речі якщо переробити ось так то получиться hash map якщо цікаво почитай
тобі потрбіно буде запускати одну з функцій в залежності від того по якому значенню ти фільтруєш ось так
```
filterHash[selectedFilter]()
```

Reply:
але так немає сенсу використовувати hash map, це я так для прикладу, і так як ти зберігати всі 3 варанти теж не ок